src/MongoDB.Driver.Core/Core/Configuration/ConnectionString.cs

```diff
@@ -987,8 +1003,13 @@ private void ParseOption(string name, string value)
                     break;
                 case "wtimeout":
                 case "wtimeoutms":
-                    _wTimeout = ParseTimeSpan(name, value);
-                    if (_wTimeout < TimeSpan.Zero)
+                    // 0 means no timeout
+                    _wTimeout = ParseTimeSpanAndReplaceIfNeeded(
```

Comment:
it looks like we don't use this option at all. But it looks like we need to use this value since this option exists in https://github.com/mongodb/specifications/blob/master/source/uri-options/uri-options.rst
@rstam , @jyemin